cumulus/client/consensus/aura/src/collators/slot_based/mod.rs

```diff
@@ -123,39 +137,49 @@ where
 	P: Pair + 'static,
 	P::Public: AppPublic + Member + Codec,
 	P::Signature: TryFrom<Vec<u8>> + Member + Codec,
+	Spawner: SpawnNamed,
 {
 	let (tx, rx) = tracing_unbounded("mpsc_builder_to_collator", 100);
 	let collator_task_params = collation_task::Params {
-		relay_client: params.relay_client.clone(),
-		collator_key: params.collator_key,
-		para_id: params.para_id,
-		reinitialize: params.reinitialize,
-		collator_service: params.collator_service.clone(),
+		relay_client: relay_client.clone(),
+		collator_key,
+		para_id,
+		reinitialize,
+		collator_service: collator_service.clone(),
 		collator_receiver: rx,
 	};
 
 	let collation_task_fut = run_collation_task::<Block, _, _>(collator_task_params);
 
 	let block_builder_params = block_builder_task::BuilderTaskParams {
-		create_inherent_data_providers: params.create_inherent_data_providers,
-		block_import: params.block_import,
-		para_client: params.para_client,
-		para_backend: params.para_backend,
-		relay_client: params.relay_client,
-		code_hash_provider: params.code_hash_provider,
-		keystore: params.keystore,
-		para_id: params.para_id,
-		proposer: params.proposer,
-		collator_service: params.collator_service,
-		authoring_duration: params.authoring_duration,
+		create_inherent_data_providers,
+		block_import,
+		para_client,
+		para_backend,
+		relay_client,
+		code_hash_provider,
+		keystore,
+		para_id,
+		proposer,
+		collator_service,
+		authoring_duration,
 		collator_sender: tx,
-		slot_drift: params.slot_drift,
+		slot_drift,
 	};
 
 	let block_builder_fut =
 		run_block_builder::<Block, P, _, _, _, _, _, _, _, _>(block_builder_params);
 
-	(collation_task_fut, block_builder_fut)
+	spawner.spawn_blocking(
```

Comment:
Do we really need blocking on this one, given the building takes place in a diff thread anyway ?https://github.com/paritytech/polkadot-sdk/blob/bd0d0cde53272833deaaa0bcaddf95fba290ea77/substrate/client/basic-authorship/src/basic_authorship.rs#L289

Reply:
You could take a peek at https://grafana.teleport.parity.io/goto/utz46S7Ng?orgId=1 to see what is blocking/async and how much it uses the CPU.

<img width="1694" alt="Screenshot 2024-11-20 at 18 37 43" src="https://github.com/user-attachments/assets/9a4fdcc5-3893-496a-9a57-f4451dfd5e2a">

Reply:
Just because it doesn't use the CPU 100%, doesn't mean it should not be blocking? Basically anything that is doing runtime calls should be running on a blocking task, because you could block the thread.

Reply:
Totally agree on that, the measured poll duration for this task is on average: 14ms, with a max of 21ms and is polled once per second on average. I think this is not that much, but it makes sense for this task to not be delayed/blocked by other things running on the thread.

Reply:
My argument is more that this task should not block other tasks. However, you are also right that this task should not need to wait for others :D In the end the operating systems are quite good at handling threads as well. So, I don't think this will have any negative impact.